src/side-panel/side-panel.js

```diff
@@ -0,0 +1,187 @@
+  width = 3,
+}) => {
+  const handleToggleExpanded = useCallback(() => {
+    if (!expanded) {
+      setExpanded(true)
+    } else {
+      if (onClose) {
+        onClose()
+      }
+      setExpanded(false)
+    }
+  }, [expanded, onClose])
+
+  const icon = (
+    <Box
+      sx={{
+        p: 1,
+        mr:
+          side === 'left'
+            ? ['29px', '29px', '29px', '24px']
+            : ['22px', '22px', '22px', '17px'],
+        pl: side === 'right' ? '7px' : null,
+      }}
+    >
+      <ArrowThin
+        id='arrow'
+        sx={{
+          strokeWidth: 2,
+          width: 24,
+          height: 24,
+          transition: 'transform 0.2s',
+          transform:
+            (expanded && side === 'left') || (!expanded && side === 'right')
+              ? 'scaleX(-1)'
+              : '',
+          mb: ['-6px', '-6px', '-6px', '-4px'],
+        }}
+      />
+    </Box>
+  )
+
+  return (
+    <>
+      <SidePanelAttachment
+        width={width}
+        side={side}
+        expanded={expanded}
+        sx={{ bottom: ['18px', '18px', '18px', '16px'] }}
+      >
+        <Button
+          onClick={handleToggleExpanded}
+          prefix={side === 'left' ? icon : null}
+          suffix={side === 'right' ? icon : null}
+          size='sm'
+          sx={{
+            display: ['none', 'none', 'inline-block', 'inline-block'],
+            cursor: 'pointer',
+            color: 'primary',
+          }}
+        >
+          {expanded ? null : tooltip}
+        </Button>
+      </SidePanelAttachment>
+      <Row>
+        <Column width={width} start={1}>
+          <Box
+            sx={{
+              position: 'absolute',
+              [side]: '0px',
+              [side === 'left' ? 'right' : 'left']: [
+                `calc(${6 - width} * (100vw - 7 * 24px) / 6 + ${
+                  12 - width
+                } * 24px)`,
+                `calc(${8 - width} * (100vw - 9 * 32px) / 8 + ${
+                  12 - width
+                } * 32px)`,
+                `calc(${12 - width} * (100vw - 13 * 32px) / 12 + ${
+                  12 - width
+                } * 32px)`,
+                `calc(${12 - width} * (100vw - 13 * 48px) / 12 + ${
+                  12 - width
+                } * 48px)`,
+              ],
+              zIndex: 1000,
+              transition: 'transform 0.2s',
+              transform: expanded
+                ? 'translateX(0)'
+                : `translateX(${side === 'left' ? '-' : ''}100%)`,
+            }}
+          >
+            <Box
+              sx={{
+                px: [4, 5, 5, 6],
+                height: '56px',
+                bg: 'background',
+                [side === 'left' ? 'borderRight' : 'borderLeft']: ({
+                  colors,
+                }) => `${expanded ? 1 : 0}px solid ${colors.muted}`,
+                borderBottom: ({ colors }) =>
+                  `${expanded ? 1 : 0}px solid ${colors.muted}`,
+                transition: 'border 0.2s',
```

Comment:
consider removing this and the handling above

Reply:
Okay after playing with this a bit, I'm in inclined to leave this transition in place. The reason being that all of the other side bar borders (width) and content (opacity) have a transition states, so we should be consistent here.